OpenFOAM/bike_OpenFOAM_v11.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+
+module load OpenFOAM/11-foss-2023a
+
+which ssh &> /dev/null
+if [ $? -ne 0 ]; then
+    # if ssh is not available, set plm_rsh_agent to empty value to avoid OpenMPI failing over it
+    # that's OK, because this is a single-node run
+    export OMPI_MCA_plm_rsh_agent=''
+fi
+
+source $FOAM_BASH
+
+if [ -z $EBROOTOPENFOAM ]; then
+    echo "ERROR: OpenFOAM module not loaded?" >&2
+    exit 1
+fi
+
+# Allow users to define the WORKDIR externally (for example a shared FS for multinode runs)
+export WORKDIR="${WORKDIR:-/tmp/$USER/$$}"
+echo "WORKDIR: $WORKDIR"
+mkdir -p $WORKDIR
+cd $WORKDIR
+pwd
+
+# motorBike, 2M cells
+BLOCKMESH_DIMENSIONS="100 40 40"
+# motorBike, 150M cells
+#BLOCKMESH_DIMENSIONS="200 80 80"
+
+# X*Y*Z should be equal to total number of available cores (across all nodes)
+X=${X:-4}
+Y=${Y:-2}
+Z=${Z:-1}
+# number of nodes
+NODES=${NODES:-1}
+# total number of cores
+NP=$((X * Y * Z))
+# cores per node
+PPN=$(((NP + NODES -1)/NODES))
+
+CASE_NAME=motorBike
+
+if [ -d $CASE_NAME ]; then
+    echo "$CASE_NAME already exists in $PWD!" >&2
+    exit 1
+fi
+
+cp -r $WM_PROJECT_DIR/tutorials/incompressibleFluid/motorBike $CASE_NAME
+chmod -R u+w $CASE_NAME
+cd $CASE_NAME/$CASE_NAME
+pwd
+
+# generate mesh
+# All Foam dictionary sub entries are accssed using / (<main entry>/<sub entry>) rather than . (<main entry>.<sub entry>)
+echo "generating mesh..."
+# Needed to reduce this to a smaller value 200 million is too big for 8 processes, therefore setting to 8 million.
+foamDictionary -entry castellatedMeshControls/maxGlobalCells -set 8000000 system/snappyHexMeshDict
+foamDictionary -entry blocks -set "( hex ( 0 1 2 3 4 5 6 7 ) ( $BLOCKMESH_DIMENSIONS ) simpleGrading ( 1 1 1 ) )" system/blockMeshDict
+foamDictionary -entry numberOfSubdomains -set $NP system/decomposeParDict
+foamDictionary -entry hierarchicalCoeffs/n -set "($X $Y $Z)" system/decomposeParDict
+
+# this needs to be moved to constant/geometry and not constant/triSurface/
+cp $WM_PROJECT_DIR/tutorials/resources/geometry/motorBike.obj.gz constant/geometry/
+#surfaceFeaturesDict not available.
+# surfaceFeatures 2>&1 | tee log.surfaceFeatures 
+blockMesh 2>&1 | tee log.blockMesh
+decomposePar -copyZero 2>&1 | tee log.decomposePar
+mpirun -np $NP -npernode $PPN snappyHexMesh -parallel -overwrite 2>&1 | tee log.snappyHexMesh
```

Comment:
This assumes you have the correct number of cores, but the workload is tiny, allow oversubscription.
```suggestion
mpirun -np $NP -npernode $PPN --oversubscribe snappyHexMesh -parallel -overwrite 2>&1 | tee log.snappyHexMesh
```

Reply:
Actually, I guess it may be better to set this as an environment variable in `run.sh` rather than embed it permanently in the scripts

Reply:
```
export OMPI_MCA_rmaps_base_oversubscribe=true
```

Reply:
(I've tested that flag and it works, it would remove the need for `--oversubscribe` in the scripts)